cmd/proxy/main_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+)
+
+func TestHandleClient_Positive(t *testing.T) {
+	type subtest struct {
+		name    string
+		backend string
+
+		// One stream per query
+		streams []*MockQUICStream
+	}
+
+	subtests := []*subtest{
+		{
+			name:    "recursive_A",
+			backend: "8.8.4.4:53",
+			streams: []*MockQUICStream{
+				NewMockQUICStream(0, "ns1.com", "A", true, false),
+			},
+		},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(tt *testing.T) {
+			ms := &MockQUICSession{
+				Streams: st.streams,
+			}
+
+			ctx := context.Background()
+			l := log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
```

Comment:
Please, can you use `NewNopLogger()`? We do not need to log anything in the tests.